### PR TITLE
fix: @W-8612427@ load project, if available, to establish loginurl

### PR DESCRIPTION
**Pull request by peternhale**
### What does this PR do?

Corrects the behavior to see if the command is being run from a sfdx project dir and then load the project json to acquire login URL if present

### What issues does this PR fix or reference?
None

### CODE REVIEW

File: src/common.ts [hunk 1]
```diff
@@ -24,4 +25,17 @@ export class Common {
...
   }
+  public static async getLoginUrl(instanceUrl: Optional<string>): Promise<Optional<string>> {
+    if (instanceUrl) {
+      return instanceUrl;
+    }
+    try {
+      const project = await SfdxProject.resolve();
+      const projectJson = await project.resolveProjectConfig();
+      return getString(projectJson, 'sfdcLoginUrl', SfdcUrl.PRODUCTION);
+    } catch (err) {
```
Comment (WillieRuemmele) [inline src/common.ts:36]: would it be useful to add some logging around here? Or is the instanceUrl already being logged from something else?

File: test/common.test.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import sinon = require('sinon');
+import { expect } from '@salesforce/command/lib/test';
+import { restoreContext, testSetup } from '@salesforce/core/lib/testSetup';
+import { Common } from '../src/common';
+
+describe('common unit tests', () => {
+  const sandbox = sinon.createSandbox();
+  const $$ = testSetup();
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars,@typescript-eslint/no-explicit-any
+  let projectPath: any;
```
Comment (WillieRuemmele) [inline test/common.test.ts:17]: why is this `any`? shouldn't that be a string?

File: src/common.ts (hunk 1 above)
Comment (peternhale) [reply src/common.ts:36]: I will add some logging to record how the login URL resolves.

File: test/common.test.ts (hunk 2 above)
Comment (peternhale) [reply test/common.test.ts:17]: changing to string

File: src/common.ts [hunk 3]
```diff
@@ -24,4 +25,21 @@ export class Common {
...
+      return instanceUrl;
+    }
+    const logger = await Logger.child('Common', { tag: 'getLoginUrl' });
+    let loginUrl: string;
+    try {
+      const project = await SfdxProject.resolve();
+      const projectJson = await project.resolveProjectConfig();
+      loginUrl = getString(projectJson, 'sfdcLoginUrl', SfdcUrl.PRODUCTION);
+    } catch (err) {
+      logger.debug(`error occurred while trying to determin loginUrl: ${err.message as string}`);
```
Comment (WillieRuemmele) [inline src/common.ts:39]: nit: spelling "determin" -> "determine"

File: src/common.ts [hunk 4]
```diff
@@ -24,4 +25,21 @@ export class Common {
       });
     }
   }
+  public static async getLoginUrl(instanceUrl: Optional<string>): Promise<Optional<string>> {
```
Comment (amphro) [inline src/common.ts:28]: nit: I'm not a fan of async methods that start with `get`. I would say either remove async or call it `resolveLoginUrl`

**Review by amphro — APPROVED** [review]

**Comment by peternhale** [comment]
@amphro I had to change the logout tests. The spy on config.unset count had to be cut in half for ALL occurrences. Not sure why.

**Pull request merged**